sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java

```diff
@@ -40,21 +43,31 @@
 @ConditionalOnClass(EventHubsTemplate.class)
 @AutoConfigureAfter(AzureEventHubsAutoConfiguration.class)
 @ConditionalOnProperty(value = "spring.cloud.azure.eventhubs.enabled", havingValue = "true", matchIfMissing = true)
-@ConditionalOnAnyProperty(prefix = "spring.cloud.azure.eventhubs", name = { "connection-string", "namespace" })
+@ConditionalOnAnyProperty(prefix = "spring.cloud.azure.eventhubs", name = {"connection-string", "namespace"})
 @ConditionalOnBean(AzureEventHubsProperties.class)
 @Import({
     AzureEventHubsMessagingAutoConfiguration.EventHubsTemplateConfiguration.class,
     AzureEventHubsMessagingAutoConfiguration.ProcessorContainerConfiguration.class
 })
 public class AzureEventHubsMessagingAutoConfiguration {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureEventHubsMessagingAutoConfiguration.class);
+
     @Bean
     @ConditionalOnMissingBean
-    NamespaceProperties eventHubNamespaceProperties(AzureEventHubsProperties properties) {
+    NamespaceProperties eventHubsNamespaceProperties(AzureEventHubsProperties properties,
+                                                      ObjectProvider<ServiceConnectionStringProvider<AzureServiceType.EventHubs>> connectionStringProviders) {
         NamespaceProperties namespaceProperties = new NamespaceProperties();
-        AzurePropertiesUtils.copyAzureCommonProperties(properties, namespaceProperties);
         BeanUtils.copyProperties(properties, namespaceProperties);
         copyAzureCommonProperties(properties, namespaceProperties);
+        if (namespaceProperties.getConnectionString() == null) {
+            ServiceConnectionStringProvider<AzureServiceType.EventHubs> connectionStringProvider =
+                connectionStringProviders.getIfAvailable();
+            if (connectionStringProvider != null) {
+                namespaceProperties.setConnectionString(connectionStringProvider.getConnectionString());
+                LOGGER.warn("Event Hubs connection string is set now.");
```

Comment:
change to debug / info,` "Event Hubs connection string is set from {} now.", connectionStringProvider.getClass().getName()`

Reply:
Cool, good idea, we can avoid print the connection string and also show users more info in this way.